dotCMS/src/main/java/com/dotmarketing/filters/CMSFilter.java

```diff
@@ -139,6 +150,7 @@ public void doFilter(ServletRequest req, ServletResponse res, FilterChain chain)
 				rewrite = VirtualLinksCache.getPathFromCache(("/".equals(uri) ? "/cmsHomePage" : uri.endsWith("/")?uri.substring(0, uri.length() - 1):uri));
 			}
 			if (UtilMethods.isSet(rewrite) && rewrite.contains("//")) {
+			    response.setStatus(VANITY_URL_REDIRECT_CODE);
 				response.sendRedirect(rewrite);
```

Comment:
"Setting the status before sendRedirect() won't work as sendRedirect() would overridde it to SC_FOUND afterwards." http://stackoverflow.com/questions/9034149/httpservletresponse-sendredirect-permanent